### ISSUE TITLE
token returns null when awaited right after initializing

### ISSUE THREAD

**Issue by jteague9** [felangel/fresh#70]
Code to reproduce:

```dart
import 'package:fresh/fresh.dart';

class InMemoryDelayed extends InMemoryTokenStorage<String> {
  @override
  Future<String?> read() async {
    await Future<dynamic>.delayed(Duration.zero);
    return super.read();
  }
}

class Foo with FreshMixin<String> {
  Foo(InMemoryDelayed tokenStorage) {
    this.tokenStorage = tokenStorage;
  }
}

void main() async {
  var storage = InMemoryDelayed();
  await storage.write('token');
  var foo = Foo(storage);
  var token = await foo.token;
  print(token); // prints null, but should print 'token'
}
```

The `token` method is not waiting for the first read from storage to complete. 

[https://github.com/felangel/fresh/blob/master/packages/fresh/lib/src/fresh.dart#L110](https://github.com/felangel/fresh/blob/master/packages/fresh/lib/src/fresh.dart#L110) this line can be changed to 
`await authenticationStatus.firstWhere((status) => status != AuthenticationStatus.initial);`

I can put up a PR if you think this fix is appropriate.

**Comment by lyio** [comment]
I ran into the same problem yesterday.

**Comment by felangel** [comment]
Closed by #96
